components/scream/src/physics/rrtmgp/scream_rrtmgp_interface.cpp

```diff
@@ -304,13 +328,175 @@ namespace scream {
+                        // Check cldx in level above and see if it satisfies conditions to create a cloudy subcolumn
+                        if (cldx(icol,ilay-1,igpt) > 1.0 - cldf(icol,ilay-1)) {
+                            // Cloudy subcolumn above, use same random number here so that clouds in these two adjacent
+                            // layers are maximimally overlapped
+                            cldx(icol,ilay,igpt) = cldx(icol,ilay-1,igpt);
+                        } else {
+                            // Cloud-less above, use new random number so that clouds are distributed
+                            // randomly in this layer. Need to scale new random number to range
+                            // [0, 1.0 - cldf(ilay-1)] because we have artifically changed the distribution
+                            // of random numbers in this layer with the above branch of the conditional,
+                            // which would otherwise inflate cloud fraction in this layer.
+                            cldx(icol,ilay,igpt) = cldx(icol,ilay  ,igpt) * (1.0 - cldf(icol,ilay-1));
+                        }
+                    }
+                });
+            }
+
+            // Use cldx array to create subcolumn mask
+            parallel_for(Bounds<3>(ngpt,nlay,ncol), YAKL_LAMBDA(int igpt, int ilay, int icol) {
+                if (cldx(icol,ilay,igpt) > 1.0 - cldf(icol,ilay)) {
+                    subcolumn_mask(icol,ilay,igpt) = 1;
+                } else {
+                    subcolumn_mask(icol,ilay,igpt) = 0;
+                }
+            });
+            return subcolumn_mask;
+        }
+
+
+        OpticalProps2str get_subsampled_clouds(
+                const int ncol, const int nlay, const int nbnd, const int ngpt,
+                OpticalProps2str &cloud_optics, GasOpticsRRTMGP &kdist, real2d &cld, real2d &p_lay) {
+            // Initialized subsampled optics
+            OpticalProps2str subsampled_optics;
+            subsampled_optics.init(kdist.get_band_lims_wavenumber(), kdist.get_band_lims_gpoint(), "subsampled_optics");
+            subsampled_optics.alloc_2str(ncol, nlay);
+            // Check that we do not have clouds with no optical properties; this would get corrected
+            // when we assign optical props, but we want to use a "radiative cloud fraction"
+            // for the subcolumn sampling too because otherwise we can get vertically-contiguous cloud
+            // mask profiles with no actual cloud properties in between, which would just further overestimate
+            // the vertical correlation of cloudy layers. I.e., cloudy layers might look maximally overlapped
+            // even when separated by layers with no cloud properties, when in fact those layers should be
+            // randomly overlapped.
+            auto cldfrac_rad = real2d("cldfrac_rad", ncol, nlay);
+            memset(cldfrac_rad, 0.0);  // Start with all zeros
```

Comment:
Curious (as it's not a perf issue): isn't yakl already init-ing to 0? Or is it leaving memory uninitialized?

Reply:
Good question. In reading through the YAKL source, it looks like "no", the constructor looks like it just allocates the memory and sets up the dims. But I could be missing something.

Reply:
It's not important, I was just wondering (since Kokkos always inits to 0, unless you use a special constructor that leaves mem uninited). Thanks.